_blog/2024-01-18-parsing-json-rust.mdx

```diff
@@ -0,0 +1,229 @@
+
+```bash
+cargo add serde-json
+```
+
+It requires no extra dependencies to use and is often recommended alongside `serde` when you need access to idiomatic manipulation of raw JSON values. `serde-json` also has support for `no_std` by allowing you to turn off the default `std` feature and enabling `alloc` instead. 
+
+In terms of performance, `serde-json` itself is not slow by any means. However, it is slower than some of the other JSON libraries on this list. This is primarily due to being optimised for non-parallelized CPU usage. Particularly if you are able to access a modern x86 CPU, you may want to read on to find out more about some of the better-performing options. However, this crate is also the most well used and supported within the Rust community, so if you are having issues with it then it is easy to find assistance!
+
+### simd-json
+
+`simd-json` is a Rust port of the `simdjson` C++ JSON parser, with `serde` compatibility built in. As the name states, this library uses SIMD - short for Single Instruction Multiple Data. This is a technique used to be able to process multiple data points with parallel processing, making it significantly faster! As a caveat however, it requires that your system is x86 capable and during runtime it will select the best SIMD feature set for performance. If no feature sets are available there is also an unoptimised Rust implementation, but in the documentation it mentions that it should not be relied on. 
+
+Generally speaking, although this library is quite fast, it should be noted that there is quite a lot of **unsafe** code in this crate due to it being a port of a C++ crate. This is not to say that you shouldn’t use it, but rather to use it with caution (as the crate says). In spite of this, there’s a section on safety that details how best practices (like unit testing) are upheld to make sure the crate is as safe to use as possible.
+
+It should also be mentioned that for best performance, it’s typically best to enable the `jemalloc` or `mimalloc` features to be able to make the most of the library. 
+
+Generally, the API for `simd-json` is the same as `serde-json`, so if you want to switch at any point then generally you should not have any problems doing so.
+
+### sonic-rs
+
+`sonic-rs` is a Rust implementation of JSON manipulation with SIMD functionality. This library also has a counterpart library in C++ and Go! Although it used to require the Rust nightly toolchain, it supports stable Rust. Similarly to `simd-json`, it also requires x86 CPU architecture to function at full capacity. 
+
+To use `sonic-rs`, you need to enable the following compiler option in rustc when running your program, like so:
+
+```rust
+rustc -C target-cpu=native
+```
+
+However, if you’re like most people using Cargo you probably want to use `cargo run`. As in the example, you can create a config at `.cargo/config` and then add the following:
+
+```rust
+[build]
+rustflags = ["-C", "target-cpu=native"]
+```
```

Comment:
nit: wondering if this applies for `simd-json` too. Any idea?

Reply:
It only states that native compilation gives the best performance.

I'll add this in since this seems to be mostly the same thing